snap/info_snap_yaml_test.go

```diff
@@ -1922,5 +1931,43 @@ system-usernames:
 `)
 	_, err := snap.InfoFromSnapYaml(y)
 	c.Assert(err, NotNil)
-	c.Assert(err, ErrorMatches, `"foo" has attribute that is empty string`)
+	c.Assert(err, ErrorMatches, `system username "foo" has an empty attribute key`)
+}
+
+func (s *YamlSuite) TestSnapYamlSystemUsernamesParsingBadAttrKeyNonString(c *C) {
+	y := []byte(`name: binary
+version: 1.0
+system-usernames:
+  foo:
+    scope: shared
+    1: bar
+`)
+	_, err := snap.InfoFromSnapYaml(y)
+	c.Assert(err, NotNil)
```

Comment:
Nitpick^99: This is slightly redundant, ErrorMatches in the next line will also check this.

Reply:
I can cleanup these in a follow up